src/__tests__/basic.test.tsx

```diff
@@ -0,0 +1,29 @@
+import * as React from 'react'
+import IdealImage from '../../'
+
+interface Props {}
+interface State {}
+
+export default class Application extends React.Component<Props, State> {
+  render() {
+    return (
+      <IdealImage
+        theme={{placeholder: {filter: 'blur(3px)'}}}
+        srcSet={[{src: 'some-src.jpg', width: 3500}]}
+        placeholder={{color: '#FFFFFF'}}
+        shouldAutoDownload={() => true}
+        loader="image"
+        getUrl={srcType => srcType.src}
```

Comment:
this should throw an error, because `srcType.src` is `string?`, where is  `getUrl` is `(...) => string`

Reply:
Actually it does

```
src/__tests__/basic.test.tsx:16:9 - error TS2326: Types of property 'getUrl' are incompatible.
  Type '(srcType: SrcType) => string | undefined' is not assignable to type '((srcType: SrcType) => string) | undefined'.
    Type '(srcType: SrcType) => string | undefined' is not assignable to type '(srcType: SrcType) => string'.
      Type 'string | undefined' is not assignable to type 'string'.
        Type 'undefined' is not assignable to type 'string'.
16         getUrl={srcType => srcType.src}
```